mcp-server/data/prompts.yml

```diff
@@ -53,5 +53,38 @@ initial_instructions: |
     * findShape(predicate: (shape: Shape) => boolean, root: Shape | null = null): Shape | null
       If no root is provided, search globally (in all pages).
     * findShapes(predicate: (shape: Shape) => boolean, root: Shape | null = null): Shape[]
-  
+
+  CRITICAL WORKFLOW - Follow these steps when working with Penpot designs:
+    1. ALWAYS use `penpotUtils` helper functions - NEVER reimplement shape searching
+    2. When working with unknown designs, start with `penpotUtils.shapeStructure(penpot.root, 3)` to understand hierarchy
+    3. Use `penpotUtils.findShapes()` or `penpotUtils.findShape()` with predicates to locate elements
+
+  COMMON TASKS - Quick Reference (ALWAYS use penpotUtils for these):
+
+    Find all images (including shapes with image fills):
+      const images = penpotUtils.findShapes(
+        shape => shape.type === 'image' || shape.fills?.some(fill => fill.fillImage),
+        penpot.root
+      );
+
+    Find text elements:
+      const texts = penpotUtils.findShapes(shape => shape.type === 'text', penpot.root);
+
+    Find shapes by name:
+      const shape = penpotUtils.findShape(shape => shape.name === 'MyShape');
+
+    Get structure of current selection:
+      const structure = penpotUtils.shapeStructure(penpot.selection[0]);
+
+    Find shapes in current selection/board:
+      const shapes = penpotUtils.findShapes(predicate, penpot.selection[0] || penpot.root);
+
+  CRITICAL WARNINGS ABOUT IMAGES:
+    *  Images appear in TWO ways in Penpot:
+        1. As standalone `Image` shapes (shape.type === 'image')
+        2. As fills on other shapes (Rectangle, Ellipse, etc.) via `fillImage` property
+    *  Most images in designs are rectangles/shapes with image fills, NOT Image type shapes!
+    *  To find ALL images, you MUST check both: shape.type === 'image' OR shape.fills with fillImage
+    *  NEVER assume shape.type === 'image' is the only way images appear
```

Comment:
I don't thinkt his needs to be a "critical warning". The model did not explicitly have this information before, and adding it plainly should suffice. Let's reserve the strong words and all-caps for cases where a normal tone definitely does not work.

Also, given the information above, the last point seems redundant. Was it really necessary to add it?